src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -243,51 +298,70 @@ class SidebarPanels extends React.Component<Props, State> {
 
         const isMetadataSidebarRedesignEnabled = isFeatureEnabled(features, 'metadata.redesign.enabled');
         const isMetadataAiSuggestionsEnabled = isFeatureEnabled(features, 'metadata.aiSuggestions.enabled');
-        const { shouldBeDefaultPanel: shouldBoxAIBeDefaultPanel, showOnlyNavButton: showOnlyBoxAINavButton } =
-            getFeatureConfig(features, 'boxai.sidebar');
-
-        const canShowBoxAISidebarPanel = hasBoxAI && !showOnlyBoxAINavButton;
+        const { shouldBeDefaultPanel: shouldBoxAIBeDefaultPanel } = getFeatureConfig(features, 'boxai.sidebar');
+
+        const hasCustomPanels = customPanels && customPanels.length > 0;
+
+        // Build eligibility for custom panels
+        const customPanelEligibility = {};
+        if (hasCustomPanels && customPanels) {
+            customPanels.forEach(panel => {
+                const canShowPanel = !panel.isDisabled;
+                if (canShowPanel) {
+                    customPanelEligibility[panel.path] = true;
+                }
+            });
+        }
```

Comment:
```suggestion
        if (hasCustomPanels) {
            customPanels.forEach(panel => {
                const canShowPanel = !panel.isDisabled;
                if (canShowPanel) {
                    customPanelEligibility[panel.path] = true;
                }
            });
        }
```

Reply:
Remove  && customPanels will leads to `flow check` error:  `Cannot call customPanels.forEach because property forEach is missing in undefined [1].`